### PR TITLE
[JAX] triangle attention

### PR DESCRIPTION
**Pull request by mariogeiger**
- VJP rule
- Call cuda code only if we are running on a GPU

### CODE REVIEW

File: cuequivariance_jax/cuequivariance_jax/triangle/_triangle_attention.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    if precision is None:
+        precision = jax.lax.Precision.DEFAULT
+
+    return fwd_p.bind(q, k, v, mask, bias, scale=scale, precision=precision)
+
+
+def triangle_attention_fwd(q, k, v, mask, bias, scale, precision=None):
+    a, lse, amax = fwd_p.bind(q, k, v, mask, bias, scale=scale, precision=precision)
+    residuals = (a, lse, q, k, v, mask, bias)
+    return (a, lse, amax), residuals
```
Comment (hsadasivan) [inline cuequivariance_jax/cuequivariance_jax/triangle/_triangle_attention.py:251]: why are we returning residuals?

File: cuequivariance_jax/cuequivariance_jax/triangle/_triangle_attention.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+
+def triangle_attention_bwd(scale, precision, residuals, cotangents):
+    a, lse, q, k, v, mask, bias = residuals
+    da, dlse, damax = cotangents
+
+    dq, dk, dv, dbias = bwd_p.bind(
+        da, a, lse, q, k, v, mask, bias, scale=scale, precision=precision
```
Comment (hsadasivan) [inline cuequivariance_jax/cuequivariance_jax/triangle/_triangle_attention.py:259]: For consistency with torch, isn't it better to do bias,mask rather than mask,bias ?

File: cuequivariance_jax/tests/triangle_attention_test.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+def test_basic_functionality(platform):
+    """Basic test to ensure the function works."""
+    require_platform(platform)
+
+    q, k, v, mask, bias, scale = create_test_data(platform)
+
+    def fn(q, k, v, mask, bias):
+        return cuex.triangle_attention(q, k, v, mask, bias, scale)
+
+    output, lse, amax = fn(q, k, v, mask, bias)
```
Comment (hsadasivan) [inline cuequivariance_jax/tests/triangle_attention_test.py:102]: why dont we have more elaborate parametrized tests for triattn in jax like in trimul (shapes, precision etc.)
